### PR TITLE
Clean up path handling

### PR DESCRIPTION
**Pull request by mpcusack-color**
Cleans up and more heavily comments environment file finding code in clrenv.

Biggest functional change is to not automatically look for overlap/user files if they are not specified explicitly by env vars. This wasn't being used (we set it explicitly in `setenv.sh`) and didn't work (we now store these under `local/` where this wasn't looking).

### CODE REVIEW

File: clrenv/path.py [hunk 1]
```diff
@@ -1,32 +1,74 @@
...
+    result = []
+    if not paths:
+        return result
+
+    for path in paths.split(":"):
+        resolved = _resolve_path(path)
+        if resolved.is_file():
+            result.append(resolved)
+        elif DEBUG_MODE:
+            logging.warning(f'Could not find "{path}" {resolved}, ignoring it.')
```
Comment (kkchau) [inline clrenv/path.py:42]: I think this should be `logger`?

File: clrenv/path.py [hunk 2]
```diff
@@ -1,32 +1,74 @@
...
+    return Path(path).expanduser().absolute()
+
+
+def _resolve_paths(paths):
+    """Resolves a user provided, colon seperated, list of paths.
+
+    Will drop values that do not exist."""
+    result = []
+    if not paths:
+        return result
```
Comment (kkchau) [inline clrenv/path.py]: nit: Might be a way to only need one `return result` statement, i.e. gracefully manage `paths=''` or `paths=None` in L41?

**Review by kkchau — APPROVED** [review]

File: clrenv/path.py [hunk 3]
```diff
@@ -1,32 +1,74 @@
...
-    path = _recursively_find_file_path(name)
+def _resolve_path(path):
+    """Resolves a user provided path. Returns None for empty string.
+
+    Does not confirm the file exists.
+    """
     if not path:
-        raise Exception("%s could not be located." % name)
-    return path
+        return
```
Comment (DavidCain) [inline clrenv/path.py:28]: Nit: When a method normally has a return value, favor `return None`.

While a plain `return` _implicitly_ returns the value `None`, it does imply to most casual users that the method is not meant to return anything (in effect, `void`, if Python had such a concept).

File: clrenv/path.py (hunk 1 above)
Comment (DavidCain) [reply clrenv/path.py:42]: You _can_ use raw `logging` invocations if desired, though a `logger` instance is generally preferable.

File: clrenv/path.py [hunk 4]
```diff
@@ -1,32 +1,74 @@
...
+            logging.warning(f'Could not find "{path}" {resolved}, ignoring it.')
+    return result
+
+
+def _find_in_cwd_or_parents(name):
+    """Finds a file with the given name starting in the cwd and working up to root."""
+    for parent in (Path().absolute() / name).parents:
+        path = parent / name
+        if path.is_file():
+            return path
```
Comment (DavidCain) [inline clrenv/path.py:51]: Can we explicitly `return None` if no files found?

**Review by DavidCain — APPROVED** [review]
Looks good to me, thanks for the extra documentation!

**Comment by mpcusack-color** [comment]
Many of these changes made it into the rewrite. Closing

**Pull request closed without merging**